libs/core-ui/src/lib/util/calculateBoxData.ts

```diff
@@ -1,44 +1,59 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+import { ErrorCohort } from "../Cohort/ErrorCohort";
 import { IHighchartBoxData } from "../Interfaces/IHighchartBoxData";
 
-export function calculateBoxData(data: number[]): IHighchartBoxData {
-  const q1 = getPercentile(data, 25);
-  const median = getPercentile(data, 50);
-  const q3 = getPercentile(data, 75);
-  const iqr = q3 - q1;
-  const lowerFence = q1 - iqr * 1.5;
-  const upperFence = q3 + iqr * 1.5;
-  const outliers = [];
+export function calculateBoxPlotDataFromErrorCohort(
+  errorCohort: ErrorCohort,
+  index: number,
+  key: string
+) {
+  // key is the identifier for the column (e.g., probability)
+  return calculateBoxPlotData(
+    errorCohort.cohort.filteredData.map((dict) => dict[key]),
+    index
+  );
+}
 
-  for (const datum of data) {
-    if (datum < lowerFence || datum > upperFence) {
-      outliers.push(datum);
-    }
-  }
+export function calculateBoxPlotData(
+  data: number[],
+  index?: number
+): IHighchartBoxData {
+  data.sort((number1: number, number2: number) => {
+    return number1 - number2;
+  });
+  const firstQuartile = getPercentile(data, 25);
+  const median = getPercentile(data, 50);
+  const thirdQuartile = getPercentile(data, 75);
+  const interquartileRange = thirdQuartile - firstQuartile;
+  // calculate fences as min and max allowed values to be inside the box
+  const lowerFence = firstQuartile - interquartileRange * 1.5;
```

Comment:
minor nit (move the *1.5 up):
```
const interquartileRange = (thirdQuartile - firstQuartile) * 1.5;
const lowerFence = firstQuartile - interquartileRange;
const upperFence = thirdQuartile + interquartileRange;
```

Reply:
Addressed locally. Will be pushed shortly :-)